src/cloudant/database.py

```diff
@@ -77,6 +79,32 @@ def new_document(self):
         super(CloudantDatabase, self).__setitem__(doc['_id'], doc)
         return doc
 
+    def design_documents(self):
+        """
+        _design_documents_
+
+        Return the raw JSON content of the design documents for this database
+
+        """
+        url = posixpath.join(self.database_url, '_all_docs')
+        query = "startkey=\"_design\"&endkey=\"_design0\"&include_docs=true"
+        resp = self._r_session.get(url, params=query)
```

Comment:
I notice you aren't using `raise_for_status` -- doing so might offer clearer exceptions in the case of errors, than a JSON content exception or KeyError afterwards. Same for requests below.